src/services/arweave/WarpDataProvider.ts

```diff
@@ -9,8 +10,14 @@ import {
 export class WarpDataProvider implements SmartweaveContractSource {
   private _warp: Warp;
 
-  constructor() {
-    this._warp = WarpFactory.forMainnet();
+  constructor(arweave: Arweave) {
+    this._warp = WarpFactory.forMainnet(
+      {
+        ...defaultCacheOptions,
+      },
+      true,
+      arweave,
+    );
```

Comment:
nit: please add a comment here as to why we are using the arweave gateway

Reply:
added comment